pkg/util/promtest/histogram.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package promtest
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// IsNativeHistogram checks if a gathered metric has positive or negative spans.
+//
+// This is possible because native histograms without any observations and zero_threshold of 0
+// use a no-op span (offset 0, length 0).
+func IsNativeHistogram(g prometheus.Gatherer, metricName string) error {
```

Comment:
nit: technically the DTO can have classic buckets as well, so maybe call this HasNativeHistogram ? No strong feelings on this.